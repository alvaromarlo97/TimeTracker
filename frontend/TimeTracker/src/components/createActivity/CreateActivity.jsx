/* eslint-disable no-underscore-dangle */
/* eslint-disable global-require */
/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-no-bind */
import React, { useState } from 'react';
import {
  ScrollView,
  TextInput,
  TouchableOpacity,
  View,
  Text,

} from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { CreateNewActivity } from '../../redux/actions/userCreators';
import styles from './CreateActivity.style';

export default function CreateActivity({ navigation }) {
  const dispatch = useDispatch();
  const [color, setcolor] = useState('blue');
  const [activityName, setActivityName] = useState('');
  const userId = useSelector(({ loggedUser }) => loggedUser?.user?._id);
  function submitActivity(name, co) {
    if (activityName) {
      dispatch(CreateNewActivity(userId, {
        activityName: name,
        color: co,
      }));
      navigation.pop();
    }
  }
  return (
    <ScrollView>
      <View style={styles.textContainer}>
        <TouchableOpacity style={[styles.selectedColor, { backgroundColor: color }]} />
        <TextInput
          placeholder="Activity Name"
          autoCorrect={false}
          autoCapitalize="words"
          style={styles.activity}
          onChangeText={(text) => setActivityName(text)}
          value={activityName}

        />
      </View>
      <TouchableOpacity style={styles.button} onPress={() => submitActivity(activityName, color)}>
        <Text>Save</Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.red} onPress={() => setcolor('red')} />
      <TouchableOpacity style={styles.green} onPress={() => setcolor('green')} />
      <TouchableOpacity style={styles.black} onPress={() => setcolor('black')} />

    </ScrollView>
  );
}
