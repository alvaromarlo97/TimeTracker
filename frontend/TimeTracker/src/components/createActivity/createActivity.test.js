import React from 'react';
import { render } from '../../../utils/testUtils/test.utils';
import CreateActivity from './CreateActivity';

const navigation = {
  navigate: jest.fn(),
};

describe('Given a Character Component', () => {
  describe('When it is rendered', () => {
    let screen;
    beforeEach(() => {
      screen = render(<CreateActivity navigation={navigation} />);
    });
    test('Then screen should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });
});
