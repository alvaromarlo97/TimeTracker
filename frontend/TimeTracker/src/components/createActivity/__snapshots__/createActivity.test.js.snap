// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Character Component When it is rendered Then screen should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#262155",
      "display": "flex",
      "justifyContent": "center",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#262155",
          "borderBottomLeftRadius": 25,
          "borderBottomRightRadius": 25,
          "display": "flex",
          "elevation": 20,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingLeft": 50,
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 0.51,
          "shadowRadius": 13.16,
        },
        Object {
          "shadowColor": "#FF3B30",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FF3B30",
          "borderRadius": 50,
          "height": 50,
          "marginRight": 50,
          "opacity": 1,
          "width": 50,
        }
      }
    />
    <TextInput
      autoCapitalize="words"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Activity Name"
      placeholderTextColor="#FCFFFF"
      style={
        Object {
          "alignItems": "center",
          "color": "#FCFFFF",
          "display": "flex",
          "flex": 1,
          "fontFamily": "Avenir-Medium",
          "fontSize": 25,
          "height": 200,
          "width": "100%",
        }
      }
      value=""
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FD9F32",
        "borderRadius": 10,
        "display": "flex",
        "height": 69,
        "justifyContent": "center",
        "margin": 30,
        "opacity": 1,
        "shadowColor": "#000",
        "width": 175,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FCFFFF",
          "fontFamily": "Avenir-Medium",
          "fontSize": 25,
        }
      }
    >
      Add
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FF3B30",
          "borderRadius": 50,
          "height": 50,
          "margin": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FF9500",
          "borderRadius": 50,
          "height": 50,
          "margin": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FFCC00",
          "borderRadius": 50,
          "height": 50,
          "margin": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4CD964",
          "borderRadius": 50,
          "height": 50,
          "margin": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#5AC8FA",
          "borderRadius": 50,
          "height": 50,
          "margin": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#007AFF",
          "borderRadius": 50,
          "height": 50,
          "margin": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#5856D6",
          "borderRadius": 50,
          "height": 50,
          "margin": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#FF2D55",
          "borderRadius": 50,
          "height": 50,
          "margin": 20,
          "opacity": 1,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;
